app/controllers/activity_forms_controller.rb

```diff
@@ -22,6 +22,12 @@ def show
     when :linked_activity
       skip_step unless policy(@activity).update_linked_activity?
       @options = linkable_activities_options(@activity)
+    when :purpose
+      @title_hint = t("form.hint.activity.title", level: t("page_content.activity.level.#{@activity.level}"))
```

Comment:
I wonder if these belong in some helper or presenter, this controller is particularly hefty, and this logic is very view-based 🤔

Reply:
Yeah I wondered about this too. We don't seem to have any hint methods in the `ActivityPresenter` or `ActivityHelper`, but we do have `linkable_activity_select_label` in the presenter, so I think it could go in the same place

Reply:
I've pushed this as a fixup commit for now: https://github.com/UKGovernmentBEIS/beis-report-official-development-assistance/pull/2031/commits/dc54e4f16ae59438c8d7bfaeb3c1b2cf9bbe3a74. What do you think of this? If it looks good, I'm squash and merge

Reply:
Hmm, I think that's better? What do you think?

They should maybe be `linkable_activity_title` and `linkable_activity_hint` though?

Reply:
This is for just the title and description field for activities (it's not related to linking - confusing, I know!)

I think it's better to shift this detail over to the presenter, yeah, and I think the presenter unit tests are more direct than testing the controller response

Reply:
Oh sorry, I didn't mean linking, I have linking on the brain! I meant the purpose step. Should it be `purpose_title` `purpose_hint`

Reply:
Ah! So this step, confusingly, contains two fields, neither of which are `purpose`. It's a single step that includes fields for `title` and `description`, and what what we're doing here is determining the hint for the `title` field and the hint for the `description` field

Reply:
👍 I'm happy with everything if you are!